### PR TITLE
Windows Domain Controller: Read Only

### PR DESCRIPTION
**Pull request by tombastianello**
##### SUMMARY
This change adds the option for users to enter "read_only: True" when promoting a domain controller, this will then create a read only domain controller rather than writable.

This feature could be very useful for a lot of large companies.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
win_domain_controller

##### ANSIBLE VERSION
2.5devel

### CODE REVIEW

File: lib/ansible/modules/windows/win_domain_controller.py [hunk 1]
```diff
@@ -48,6 +48,12 @@
   local_admin_password:
     description:
       - password to be assigned to the local C(Administrator) user (required when C(state) is C(member_server))
+  read_only:
```
Comment (bcoca) [inline lib/ansible/modules/windows/win_domain_controller.py:51]: missing version_added, also from examples (nice to add in the 'name' field)

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_domain_controller.py (hunk 1 above)
Comment (tombastianello) [reply lib/ansible/modules/windows/win_domain_controller.py:51]: changes added

**Review by nitzmahone — CHANGES_REQUESTED** [review]
A couple of things:
* The `sitecode` arg should probably be named `site_name` to use the proper terminology that Windows admins are used to.
* There are a number of unrelated formatting changes that make review more difficult and mess up change attribution. Please limit changes to only the lines that affect the new feature.

**Comment by dagwieers** [comment]
Beware that the deadline for getting new features/modules accepted in [Ansible v2.5 is nearing](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_5.rst), it is set to either 2018-01-15 or 2018-01-31. If you are blocked, or you need feedback, please discuss on IRC channel #ansible-windows or add a comment to [the Windows Working Group meeting agenda](https://github.com/ansible/community/issues/195) to get it resolved.

**Comment by jborean93** [comment]
Closing this PR in favour of https://github.com/ansible/ansible/pull/33850 as there was no response from the original author.

**Pull request closed without merging**